src/pages/word-duel.jsx

```diff
@@ -5,37 +5,92 @@ import Board from '../components/board';
 
 const MAX_GUESSES = 6;
 
+const EvaluationColor = {
+    Gray: "gray",
+    Yellow: "yellow",
+    Green: "green"
+}
+
 export default function WordDuel() {
     // const [ word, setWord ] = useState(generateNewWord())
-    const word = "awash";
-    const [ guesses, setGuesses ] = useState([]); 
-    const [ error, setError ] = useState("");
+    const word = "stink";
+    const [guesses, setGuesses] = useState([]);
+    const [evaluations, setEvaluations] = useState([]);
+    const [gameOver, setGameOver] = useState(false);
+    const [message, setMessage] = useState("");
+
     function checkGuess(guess) {
         if (guess.length !== word.length) {
-            setError("Guess wrong length!");
+            setMessage("Guess wrong length!");
+            return false;
+        }
+
+        if (guesses.length >= MAX_GUESSES) {
+            setMessage("Guess limit reached.");
+            setGameOver(true);
             return false;
         }
 
-        setError("");
-        setGuesses([ ...guesses, guess ])
+        setMessage("");
+        setGuesses([...guesses, guess]);
+        evaluateGuess(guess);
         return true;
     }
 
+    /* 
+    TODO: These are very rudamentary rule but they are sufficient for now 
+    There are some corner cases that can occur when there are repeated letters in a guess
+    Case 1: Guess contains multiples of a letter where only 1 is in the word and both "miss"
+        Actual: Both letters are yellow
+        Expected: One letter should be yellow (the first one) and the other should be gray
+    Case 2: Guess contains multiples of a letter where only 1 is in the word where 1 "hits" and 1 "misses"
+        Actual: "Miss" is yellow and "hit" is green
+        Expected: "Miss" is gray and "hit" is green
+    Need to devise a way to make sure each letter is only used once when evaluating a guess
+    */
+    function evaluateGuess(guess) {
+        const evaluation = guess.split("").map((letter, idx) => {
+            if (letter === word[idx]) {
+                return EvaluationColor.Green;
+            }
+
+            if (word.includes(letter)) {
+                return EvaluationColor.Yellow;
+            }
+
+            return EvaluationColor.Gray;
+        }, []);
```

Comment:
The `[]` would be needed for a call to `reduce` as the initial accumulator, but not for a map.  When I wrote up that example, I did it as a `reduce` before realizing I could just use a `map` but clearly didn't clean up my code.

tl;dr you don't need this empty `[]` as the second argument, my bad

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Reduce

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/Map